driver/driver.feature

```diff
@@ -0,0 +1,1265 @@
+#    Given connection does not have database: created-after-write
+#    Given connection does not have database: created-after-read
+#    When connection open schema transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-schema
+#    Then connection does not have database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open write transaction for database: typedb
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open write transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-write
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open read transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-read
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection has database: created-after-read
+
+
+  Scenario: Driver processes transaction commit errors correctly
+    Given connection open schema transaction for database: typedb
+    When typeql schema query
+      """
+      define attribute name;
+      """
+    Then transaction commits; fails
+
+
+  # TODO: check errors on transaction commits with callbacks (on_close) set!
+
+  ###########
+  # QUERIES #
+  ###########
+
+  Scenario: Driver processes ok query answers correctly
+    Given connection open schema transaction for database: typedb
+    When get answers of typeql schema query
+      """
+      define entity person;
+      """
+    Then answer type is: ok
+    Then answer type is not: concept rows
+    Then answer type is not: concept trees
+    Then result is a successful ok
```

Comment:
Needed an alternative of "get rows" for this guy.
There are generally four steps of query answer consumption:
* Get the query answer;
* (optional) Check the query answer's type;
* Unwrap the query answer to its type;
* (optional) Collect the unwrapped answer if it's an iterator

This "result is a successful ok" is specifically "as_ok()" without failures. Not sure if I need to rename it to something else, but I'm open to actions.

Reply:
I'd use `Then answer unwraps as ok/concept rows/concept trees`
We don't normally inidicate `successful` in the step, when success is the default assumption - we do mark fails though.

Reply:
Good point, fixed.